go/adbc/driver/databricks/ipc_reader_adapter.go

```diff
@@ -0,0 +1,183 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package databricks
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/ipc"
+	dbsqlrows "github.com/databricks/databricks-sql-go/rows"
+)
+
+// Check if the rows interface supports IPC streams
+type rowsWithIPCStream interface {
+	GetArrowIPCStreams(context.Context) (dbsqlrows.ArrowIPCStreamIterator, error)
+}
+
+// ipcReaderAdapter uses the new IPC stream interface for zero-copy Arrow access
```

Comment:
There is a copy when the buffers are copied from the IPC payloads into the allocated buffers. After that, the *Arrow C Data Interface* is zero-copy: user of the driver shares the pointers.

```suggestion
// ipcReaderAdapter uses the new IPC stream interface for Arrow access
```